### PR TITLE
New Serverless Pattern - EventBridge - S3 - Lambda

### PR DESCRIPTION
**Pull request by archiev4**
Issue https://github.com/aws-samples/serverless-patterns/issues/2337, if available:

Description of changes:

A new Terraform pattern which invokes a Lambda function when an S3 bucket is created. This uses Eventbridge to check for the CreateBucket API call, invokes Lambda function and notifies the user via SNS. The Lambda function used Python 3.12 and uses a zip file.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by archiev4** [comment]
Made some change to the lambda handler and example-pattern.json

File: eventbridge-s3-lambda/README.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+## Deployment Instructions
+
+1. Create a new directory, navigate to that directory in a terminal and clone the GitHub repository:
+    ``` 
+    git clone https://github.com/aws-samples/serverless-patterns
+    ```
+1. Change directory to the pattern directory:
+    ```
+    cd _patterns-model
```
Comment (ellisms) [inline eventbridge-s3-lambda/README.md:25]: Update this to the directory you created.

File: eventbridge-s3-lambda/README.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    ```
+1. From the command line, apply the configuration in the main.tf file:
+    ```
+    terraform apply
+    ```
+1. During the prompts
+    #var.region
+    - Enter a value: {enter the region for deployment}
+
+    #var.prefix {enter any name to associate with resources}
```
Comment (ellisms) [inline eventbridge-s3-lambda/README.md:39]: I was not prompted for this.

File: eventbridge-s3-lambda/README.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    - Enter a value:
+
+## Testing
+
+1. After deploying the stack, create a Subscriber for your SNS topic (For ex, your email) and confirm the subscription.
+    https://docs.aws.amazon.com/sns/latest/dg/sns-create-subscribe-endpoint-to-topic.html
+
+1. You can create a new S3 bucket using the following CLI command
```
Comment (ellisms) [inline eventbridge-s3-lambda/README.md:47]:
```suggestion
1. Create a new S3 bucket using the following CLI command
```

File: eventbridge-s3-lambda/README.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    ```
+    aws s3api create-bucket --bucket BUCKET_NAME --region REGION_NAME
+    ```
+    Note: Make sure that the region is same as the region in which you deployed the Terraform code.
+
+1. You will see that the Lambda function is invoked and a message is sent to the SNS Topic, which is then received by the subscriber.
```
Comment (ellisms) [inline eventbridge-s3-lambda/README.md:53]:
```suggestion
1. The event invokes the Lambda function and you will receive an email from the SNS Topic. 
```

File: eventbridge-s3-lambda/README.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+1. During the prompts
+    #var.region
+    - Enter a value: {enter the region for deployment}
+
+    #var.prefix {enter any name to associate with resources}
+    - Enter a value:
+
+## Testing
+
+1. After deploying the stack, create a Subscriber for your SNS topic (For ex, your email) and confirm the subscription.
```
Comment (ellisms) [inline eventbridge-s3-lambda/README.md:44]:
```suggestion
1. After deploying the stack, create a Subscriber for your Amazon SNS topic (For ex, your email) and confirm the subscription.
```

File: eventbridge-s3-lambda/README.md [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+1. You can create a new S3 bucket using the following CLI command
+    ```
+    aws s3api create-bucket --bucket BUCKET_NAME --region REGION_NAME
+    ```
+    Note: Make sure that the region is same as the region in which you deployed the Terraform code.
+
+1. You will see that the Lambda function is invoked and a message is sent to the SNS Topic, which is then received by the subscriber.
+
+## Cleanup
```
Comment (ellisms) [inline eventbridge-s3-lambda/README.md:55]: Add step to delete the SNS subscription.

**Review by ellisms — CHANGES_REQUESTED** [review]
@archiev4 Thanks for your submission. Requesting a few minor changes before moving forward.

File: eventbridge-s3-lambda/README.md [hunk 7]
```diff
@@ -0,0 +1,77 @@
+# Lambda sending a notification when a new S3 bucket is created
```
[the remark this answers is not in the document]
Comment (archiev4) [reply eventbridge-s3-lambda/README.md]: updated

File: eventbridge-s3-lambda/README.md [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+
+This pattern creates an automated monitoring system for S3 bucket create events. It defines an EventBridge rule that listens for S3 bucket creation events and triggers a Lambda function. The Lambda function processes the event details and sends a notification to an SNS topic, informing subscribers of the new bucket creation.
```
[the remark this answers is not in the document]
Comment (archiev4) [reply eventbridge-s3-lambda/README.md]: updated

File: eventbridge-s3-lambda/README.md (hunk 1 above)
Comment (archiev4) [reply eventbridge-s3-lambda/README.md]: updated

File: eventbridge-s3-lambda/README.md (hunk 2 above)
Comment (archiev4) [reply eventbridge-s3-lambda/README.md]: updated

File: eventbridge-s3-lambda/README.md (hunk 3 above)
Comment (archiev4) [reply eventbridge-s3-lambda/README.md]: updated

File: eventbridge-s3-lambda/README.md (hunk 4 above)
Comment (archiev4) [reply eventbridge-s3-lambda/README.md]: updated

File: eventbridge-s3-lambda/README.md (hunk 5 above)
Comment (archiev4) [reply eventbridge-s3-lambda/README.md]: updated

File: eventbridge-s3-lambda/README.md (hunk 6 above)
Comment (archiev4) [reply eventbridge-s3-lambda/README.md:52]: updated

**Comment by archiev4** [comment]
Made all the requested changes

**Comment by ellisms** [comment]
Thanks @archiev4 . Can you resolve each of the conversations?

**Comment by archiev4** [comment]
> Thanks @archiev4 . Can you resolve each of the conversations?

Done @ellisms

**Review by ellisms — APPROVED** [review]

**Comment by ellisms** [comment]
Thanks @archiev4 . Adding this to the pipeline.

**Pull request merged**